java/algorithm/src/main/java/org/apache/arrow/algorithm/sort/DefaultVectorComparators.java

```diff
@@ -212,7 +212,7 @@ public UInt2Comparator() {
     public int compareNotNull(int index1, int index2) {
       char value1 = vector1.get(index1);
       char value2 = vector2.get(index2);
-      return value1 - value2;
+      return (value1 & 0xffff) - (value2 & 0xffff);
```

Comment:
it looks like there is Character.compare?

Reply:
> it looks like there is Character.compare?

Thanks for your kind reminder.
The implementation of Character#compare is as follows:

```
    public static int compare(char x, char y) {
        return x - y;
    }
```
In effect, the chars x and y are first promoted to int32 (automatically, according to Java specification), and then a minus operation is performed.
So the values x and y are actually compared as signed integers, which is not what we want here.

Reply:
Could you add documentation inline about this so I don't forget the next time we look at the code?

Reply:
Sure. Thank you for the good suggestion.